pkg/build/api/v1/types.go

```diff
@@ -395,6 +399,36 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification. The hook
+// executes a command in a temporary container running the build output image,
+// immediately after the last layer of the image is commited and before the
+// image is pushed to a registry. If none of the fields are specified, the hook
+// is not executed.
+type BuildPostCommitSpec struct {
+	// Command is the command to run. It may not be specified with Script.
+	// The Docker image's entrypoint is used if this is not provided.
+	// Use Command when you need to control what is the image entrypoint.
+	// This might be needed if the image doesn't have "/bin/sh", or if you
+	// cannot afford using a shell. In all other cases, using Script will be
+	// more convenient.
+	Command []string `json:"command,omitempty" description:"command to be executed in a container running the build output image replacing the image's entrypoint"`
+	// Args is a list of arguments that are provided to either Command,
+	// Script or the Docker image's default entrypoint. In all cases, the
+	// arguments are placed immediately after the command to be run.
+	// For Script, "/bin/sh" is prepended to the argument list, so that it
```

Comment:
Regarding $0 - all of our Origin scripts today use various hacks to find out what directory the user is in in order to build other paths.  Most of them rely on $0 or $BASH_SOURCE being "self".  We should be sure the user doesn't have similar expectation when passing Script directly - i.e. using $0.  I *think* it might be ok for $0 to `/bin/sh`, but I want to be certain.